dp-hello-world-event/service/initialise.go

```diff
@@ -0,0 +1,95 @@
+package service
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-hello-world-event/config"
+	dpkafka "github.com/ONSdigital/dp-kafka/v2"
+	dphttp "github.com/ONSdigital/dp-net/http"
+)
+
+// ExternalServiceList holds the initialiser and initialisation state of external services.
+type ExternalServiceList struct {
+	HealthCheck   bool
+	KafkaConsumer bool
+	Init          Initialiser
+}
+
+// NewServiceList creates a new service list with the provided initialiser
+func NewServiceList(initialiser Initialiser) *ExternalServiceList {
+	return &ExternalServiceList{
+		HealthCheck:   false,
+		KafkaConsumer: false,
+		Init:          initialiser,
+	}
+}
+
+// Init implements the Initialiser interface to initialise dependencies
+type Init struct{}
+
+// GetHTTPServer creates an http server and sets the Server flag to true
+func (e *ExternalServiceList) GetHTTPServer(bindAddr string, router http.Handler) HTTPServer {
+	s := e.Init.DoGetHTTPServer(bindAddr, router)
+	return s
+}
+
+// GetKafkaConsumer creates a Kafka consumer and sets the consumer flag to true
+func (e *ExternalServiceList) GetKafkaConsumer(ctx context.Context, cfg *config.Config) (KafkaConsumer, error) {
+	consumer, err := e.Init.DoGetKafkaConsumer(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+	e.KafkaConsumer = true
+	return consumer, nil
+}
+
+// GetHealthCheck creates a healthcheck with versionInfo and sets teh HealthCheck flag to true
+func (e *ExternalServiceList) GetHealthCheck(cfg *config.Config, buildTime, gitCommit, version string) (HealthChecker, error) {
+	hc, err := e.Init.DoGetHealthCheck(cfg, buildTime, gitCommit, version)
+	if err != nil {
+		return nil, err
+	}
+	e.HealthCheck = true
+	return hc, nil
+}
+
+// DoGetHTTPServer creates an HTTP Server with the provided bind address and router
+func (e *Init) DoGetHTTPServer(bindAddr string, router http.Handler) HTTPServer {
+	s := dphttp.NewServer(bindAddr, router)
+	s.HandleOSSignals = false
+	return s
+}
+
+// DoGetKafkaConsumer returns a Kafka Consumer group
+func (e *Init) DoGetKafkaConsumer(ctx context.Context, cfg *config.Config) (KafkaConsumer, error) {
+	cgChannels := dpkafka.CreateConsumerGroupChannels(1)
+	kafkaOffset := dpkafka.OffsetOldest
```

Comment:
I think in other services we defined the kafkaOldest as a boolean config parameter: true=Oldest, false=Newest (e.g. https://github.com/ONSdigital/dp-hierarchy-builder/blob/develop/config/config.go#L22  ), maybe we can do the same here?

Reply:
I have now made the offset configurable.